app/code/Magento/Backend/Model/Url.php

```diff
@@ -436,6 +438,27 @@ protected function _getScope()
         return $this->_scope;
     }
 
+    /**
+     * Get scope for the url instance with provided parameter.
+     *
+     * @param $scope string
+     * @return \Magento\Store\Model\Store
+     */
+    protected function getScope($scope = null)
```

Comment:
Adding protected functions is backward incompatible and we cannot accept that to a patch release.

Also, the proposed implementation is against CQRS principle (getter should not change the state of the system)

Reply:
@sivaschenko,

> Adding protected functions is backward incompatible

What? That's something new to me. Is it documented and then announced somewhere?